client/client.go

```diff
@@ -1532,6 +1550,10 @@ func (c *Client) registerAndHeartbeat() {
 	}
 }
 
+func (c *Client) lastHeartbeat() time.Time {
+	return c.heartbeatStop.lastOk
```

Comment:
For safe concurrent access, should this be threaded through the state store the way that `setLockOk()` is? (Probably wrapped in a `getLockOk()` method?)

Reply:
fixed, I left a comment about state store error handling which I think is the right decision, but it's a little tricky

Reply:
:+1: